File: process-logo.py
from PIL import Image
import os

# Load the image
input_path = r"c:\Users\Lenovo\Downloads\findbus-main\findbus-main\public\icon.jpg"
output_path = r"c:\Users\Lenovo\Downloads\findbus-main\findbus-main\public\icon.png"

# Open the image
img = Image.open(input_path)

# Convert to RGBA for transparency support
img = img.convert("RGBA")

# Get image data
data = img.getdata()

# Create a new list to hold the modified pixels
new_data = []

# Process each pixel - replace white and near-white colors with transparency
for item in data:
    # Check if pixel is white or near-white (high RGB values)
    # Using a threshold to handle anti-aliasing and slight color variations
    if item[0] > 240 and item[1] > 240 and item[2] > 240:
        # Replace with transparent
        new_data.append((255, 255, 255, 0))
    else:
        # Keep the original pixel with full opacity
        new_data.append(item)

# Update image data
img.putdata(new_data)

# Save as PNG with high quality
img.save(output_path, "PNG", quality=95)

print(f"Successfully converted icon.jpg to icon.png")
print(f"Output saved to: {output_path}")
print(f"Image size: {img.size}")
print(f"Image mode: {img.mode}")

File: process-logo.js
import { Jimp } from 'jimp';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const inputPath = path.join(__dirname, 'public', 'icon.png');
const outputPath = path.join(__dirname, 'public', 'icon.png');

async function processLogo() {
  try {
    console.log(`Processing image: ${inputPath}`);
    
    // Read the image
    const image = await Jimp.read(inputPath);
    
    console.log(`Original size: ${image.width}x${image.height}`);
    
    // Process each pixel to remove white background
    image.scan(0, 0, image.width, image.height, (x, y, idx) => {
      const red = image.bitmap.data[idx];
      const green = image.bitmap.data[idx + 1];
      const blue = image.bitmap.data[idx + 2];
      
      // If pixel is white or near-white, make it transparent
      if (red > 240 && green > 240 && blue > 240) {
        image.bitmap.data[idx + 3] = 0; // Set alpha to 0 (transparent)
      }
    });
    
    // Save as PNG
    await image.write(outputPath);
    
    console.log(`✓ Successfully converted to ${outputPath}`);
    console.log(`✓ White background removed and replaced with transparency`);
    console.log(`✓ Output format: PNG with alpha channel`);
    
  } catch (error) {
    console.error('Error processing image:', error.message);
    process.exit(1);
  }
}

processLogo();
